src/main/groovy/com/deploygate/gradle/plugins/tasks/inputs/DeploymentConfiguration.java

```diff
@@ -13,53 +15,64 @@
-
 public abstract class DeploymentConfiguration {
 
-    /**
-     * must be an absolute path
-     */
+    /** must be an absolute path */
     @Input
     @Optional
-    @NotNull
-    public abstract Property<String> getSourceFilePath();
+    @NotNull public abstract Property<String> getSourceFilePath();
 
     @Input
     @Optional
-    @NotNull
-    public abstract Property<String> getMessage();
+    @NotNull public abstract Property<String> getMessage();
 
     @Input
-    @NotNull
-    public abstract Property<Boolean> getSkipAssemble();
+    @NotNull public abstract Property<Boolean> getSkipAssemble();
 
     @Input
     @Optional
-    @NotNull
-    public abstract Property<String> getDistributionKey();
+    @NotNull public abstract Property<String> getDistributionKey();
 
     @Input
     @Optional
-    @NotNull
-    public abstract Property<String> getDistributionReleaseNote();
+    @NotNull public abstract Property<String> getDistributionReleaseNote();
 
     @Inject
-    public DeploymentConfiguration(@NotNull ProviderFactory providerFactory, @NotNull ProjectLayout projectLayout) {
-        getSourceFilePath().set(
-                forUseAtConfigurationTime(providerFactory.environmentVariable(DeployGatePlugin.getENV_NAME_SOURCE_FILE()))
-                        .map(s -> projectLayout.getProjectDirectory().file(s).getAsFile().getAbsolutePath())
-        );
-        getMessage().set(forUseAtConfigurationTime(providerFactory.environmentVariable(DeployGatePlugin.getENV_NAME_MESSAGE())));
-        getDistributionKey().set(forUseAtConfigurationTime(providerFactory.environmentVariable(DeployGatePlugin.getENV_NAME_DISTRIBUTION_KEY())));
-        getDistributionReleaseNote().set(
-                ProviderFactoryUtils.pickFirst(
-                        forUseAtConfigurationTime(providerFactory.environmentVariable(DeployGatePlugin.getENV_NAME_DISTRIBUTION_RELEASE_NOTE())),
-                        forUseAtConfigurationTime(providerFactory.environmentVariable(DeployGatePlugin.getENV_NAME_DISTRIBUTION_RELEASE_NOTE_V1()))
-                )
-        );
+    public DeploymentConfiguration(
+            @NotNull ProviderFactory providerFactory, @NotNull ProjectLayout projectLayout) {
+        getSourceFilePath()
+                .set(
+                        forUseAtConfigurationTime(
+                                        providerFactory.environmentVariable(
+                                                DeployGatePlugin.getENV_NAME_SOURCE_FILE()))
+                                .map(
+                                        s ->
+                                                projectLayout
+                                                        .getProjectDirectory()
+                                                        .file(s)
+                                                        .getAsFile()
+                                                        .getAbsolutePath()));
```

Comment:
I feel difficult to read this code because many returns and spaces.

But never mind in this PR. 
I think that is issue to improvement for the future 👍

Reply:
Gradle lazy API force method chaining. 🤷